src/main/scala/hydrozoa/l1/multisig/tx/finalization/BloxBeanFinalizationTxBuilder.scala

```diff
@@ -0,0 +1,68 @@
+package hydrozoa.l1.multisig.tx.finalization
+
+import com.bloxbean.cardano.client.api.util.AssetUtil
+import com.bloxbean.cardano.client.quicktx.Tx
+import com.bloxbean.cardano.client.transaction.spec.Asset
+import com.bloxbean.cardano.client.transaction.spec.script.NativeScript
+import hydrozoa.infra.{force, mkBuilder}
+import hydrozoa.l1.multisig.tx.MultisigTxs.FinalizationTx
+import hydrozoa.node.server.HeadStateReader
+import hydrozoa.{AppCtx, L1Tx}
+
+import java.math.BigInteger
+import scala.jdk.CollectionConverters.*
+import scala.language.postfixOps
+
+class BloxBeanFinalizationTxBuilder(
+    ctx: AppCtx,
+    headStateReader: HeadStateReader
+) extends FinalizationTxBuilder {
+
+    private lazy val backendService = ctx.backendService
+    private lazy val builder = mkBuilder[Tx](ctx)
+
+    override def buildFinalizationDraft(
+        r: FinalizationRecipe
+    ): Either[String, FinalizationTx] =
+
+        val treasury = headStateReader.currentTreasuryRef
+        val treasuryUtxo = backendService.getUtxoService
+            .getTxOutput(treasury._1.hash, treasury._2.ix.intValue)
+            .force
+
+        // Native script
+        val Some(headNativeScript) = headStateReader.headNativeScript
+        val script = NativeScript.deserializeScriptRef(headNativeScript.bytes)
+
+        // Beacon token to burn
+        val Some(beaconTokenName) = headStateReader.beaconTokenName
```

Comment:
Technically, all of them are beacon tokens:

- the head's beacon token (prefix 4937)
- the rollout beacon tokens (prefix 7655)
- the vote beacon tokens (prefix 3477)

So, I would rename this variable to `headBeaconToken` or `treasuryBeaconToken` (your choice).

Reply:
Lol I think you've already addressed this in #73